### ISSUE TITLE
Set .Inner and .InnerDeindent to nil when shortcode is not closed

### ISSUE THREAD

**Issue by jmooring** [gohugoio/hugo#10671]
markdown

```
{{< sc >}}

Bar
```

layouts/shortcodes/sc.html

```
{{ .Inner | len }} --> 6
{{ .InnerDeindent | len }} --> 6  (\n+ \n+ 3 + \n)
```

When a shortcode is not closed, with either a closing tag or self-closing, set .Inner and .InnerDeindent to nil.

Reference <https://discourse.gohugo.io/t/42190/2>

**Comment by bep** [comment]
I suggest we set it to an empty string. `nil` prints differently in many situations, and they're for most practical applications the same.

**Comment by jmooring** [comment]
> I suggest we set it to an empty string

Works for me.

**Comment by bep** [comment]
>When a shortcode is not closed, with either a closing tag or self-closing, set .Inner and .InnerDeindent to nil.

Looking at this now, I assume your example is incorrect:

```
{{< sc >}}

Bar
```

The above is neither closing or self-closing.  See #10675

**Comment by jmooring** [comment]
> I assume your example is incorrect

The example shows what happens (verified again just now), but this would be resolved with  #10675 (i.e., "Don't do that").
